### PR TITLE
Refactor resource volumes and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by Diptipowervs** [comment]
--- PASS: TestAccIBMPIVolumebasic (112.32s)
PASS

File: ibm/service/power/ibm_pi_constants.go [hunk 1]
```diff
@@ -351,6 +359,8 @@ const (
 	PIVolumeGroupID                   = "pi_volume_group_id"
 	PIVolumeGroupAction               = "pi_volume_group_action"
 	PIVolumeOnboardingID              = "pi_volume_onboarding_id"
+	PIVolumeProvisioning              = "pi_volume_provisioning"
+	PIVolumeProvisioningDone          = "pi_volume_provisioning_done"
```
Comment (michaelkad) [inline ibm/service/power/ibm_pi_constants.go:363]: Those consts are not define properly. 
```
	helpers.PIVolumeProvisioning     = "creating"
	helpers.PIVolumeProvisioningDone = "available"
```
A state/status should be created instead. eg:
```
State_Creating = "creating"
State_Available = "available"
```

File: ibm/service/power/resource_ibm_pi_volume.go [hunk 2]
```diff
@@ -38,156 +37,156 @@ func ResourceIBMPIVolume() *schema.Resource {
...
+				Type:             schema.TypeString,
+			},
+			Arg_VolumePool: {
 				Computed:         true,
 				DiffSuppressFunc: flex.ApplyOnce,
 				Description:      "Volume pool where the volume will be created; if provided then pi_affinity_policy values will be ignored",
+				Optional:         true,
+				Type:             schema.TypeString,
 			},
 			PIAffinityPolicy: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume.go:75]: This should be rename to `Arg_AffinityPolicy`

File: ibm/service/power/resource_ibm_pi_volume.go [hunk 3]
```diff
@@ -38,156 +37,156 @@ func ResourceIBMPIVolume() *schema.Resource {
...
-				Type:             schema.TypeString,
-				Optional:         true,
 				DiffSuppressFunc: flex.ApplyOnce,
 				Description:      "Affinity policy for data volume being created; ignored if pi_volume_pool provided; for policy affinity requires one of pi_affinity_instance or pi_affinity_volume to be specified; for policy anti-affinity requires one of pi_anti_affinity_instances or pi_anti_affinity_volumes to be specified",
+				Optional:         true,
 				ValidateFunc:     validate.InvokeValidator("ibm_pi_volume", PIAffinityPolicy),
+				Type:             schema.TypeString,
 			},
 			PIAffinityVolume: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume.go:82]: This should be rename to `Arg_AffinityVolume`

File: ibm/service/power/resource_ibm_pi_volume.go [hunk 4]
```diff
@@ -38,156 +37,156 @@ func ResourceIBMPIVolume() *schema.Resource {
...
-				Type:             schema.TypeString,
-				Optional:         true,
+				ConflictsWith:    []string{PIAffinityInstance},
 				DiffSuppressFunc: flex.ApplyOnce,
 				Description:      "Volume (ID or Name) to base volume affinity policy against; required if requesting affinity and pi_affinity_instance is not provided",
-				ConflictsWith:    []string{PIAffinityInstance},
+				Optional:         true,
+				Type:             schema.TypeString,
 			},
 			PIAffinityInstance: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume.go:89]: Same as above

File: ibm/service/power/resource_ibm_pi_volume.go [hunk 5]
```diff
@@ -38,156 +37,156 @@ func ResourceIBMPIVolume() *schema.Resource {
...
-				Type:             schema.TypeString,
-				Optional:         true,
+				ConflictsWith:    []string{PIAffinityVolume},
 				DiffSuppressFunc: flex.ApplyOnce,
 				Description:      "PVM Instance (ID or Name) to base volume affinity policy against; required if requesting affinity and pi_affinity_volume is not provided",
-				ConflictsWith:    []string{PIAffinityVolume},
+				Optional:         true,
+				Type:             schema.TypeString,
 			},
 			PIAntiAffinityVolumes: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume.go:96]: Same as above

**Review by michaelkad — CHANGES_REQUESTED** [review]
Left few comments. 
- Use constant to replace string
- It is `pi_replication_enabled` instead of `replication_enabled`

File: ibm/service/power/resource_ibm_pi_volume.go (hunk 2 above)
Comment (Diptipowervs) [reply ibm/service/power/resource_ibm_pi_volume.go]: added

**Comment by ismirlia** [comment]
Closing this PR replacing it with https://github.com/powervs-ibm/terraform-provider-ibm/pull/112

**Pull request closed without merging**